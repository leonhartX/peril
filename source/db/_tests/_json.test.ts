const legitSettings = `{
  "id": 1,
  "settings": {
  },
  "rules": {
   "pull_request": "orta/peril@pr.ts",
   "issue": "orta/peril@issue.ts"
  },
  "repos" : {
    "orta/ORStackView": {
      "issue.created": "orta/peril@lock_issues.ts"
    }
  }
}`

const mockGHContents = jest.fn()
jest.mock("../../github/lib/github_helpers", () => ({ getGitHubFileContentsWithoutToken: mockGHContents }))

import { DatabaseAdaptor } from "../index"
import jsonDB from "../json"

describe("makes the right calls to GitHub", () => {
  let db: DatabaseAdaptor = null as any

  const setup = async () => {
    mockGHContents.mockImplementationOnce(() => Promise.resolve(legitSettings))

    db = jsonDB("orta/peril@settings.json")
    return db.setup()
  }

  it("with a legit stubbed JSON file", async () => {
    await setup()

    const org = await db.getInstallation(1)
    expect(org).toMatchSnapshot()
  })

  it("gets repo rules correct", async () => {
    await setup()

    const repo = await db.getRepo(1, "orta/ORStackView")
    expect(repo).toMatchSnapshot()
  })
})

it("Raises with a bad URL", async () => {
  mockGHContents.mockImplementationOnce(() => Promise.resolve(""))
  expect.assertions(1)

  try {
    const db = jsonDB("orta/other@settings.json")
    await db.setup()
  } catch (error) {
    expect(error).toMatchSnapshot()
  }
})
