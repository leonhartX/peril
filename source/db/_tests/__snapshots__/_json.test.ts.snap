// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Raises with a bad URL 1`] = `[Error: Could not find find a JSON file for Peril settings. It's likely that Peril cannot connect to orta/other@settings.json, check the logs for more info above here. You'll probably need to make changes to your  "DATABASE_JSON_FILE" in your ENV vars.]`;

exports[`makes the right calls to GitHub gets repo rules correct 1`] = `
Object {
  "fullName": "orta/ORStackView",
  "id": 111,
  "installationID": 1,
  "rules": Object {
    "issue.created": "orta/peril@lock_issues.ts",
  },
}
`;

exports[`makes the right calls to GitHub with a legit stubbed JSON file 1`] = `
Object {
  "id": 23511,
  "repos": Object {
    "orta/ORStackView": Object {
      "issue.created": "orta/peril@lock_issues.ts",
    },
  },
  "rules": Object {
    "issue": "orta/peril@issue.ts",
    "pull_request": "orta/peril@pr.ts",
  },
  "scheduler": Object {},
  "settings": Object {
    "env_vars": Array [],
    "ignored_repos": Array [],
    "modules": Array [],
  },
}
`;
